AwsCryptographicMaterialProviders/dafny/AwsCryptographyKeyStore/src/DefaultKeyStorageInterface.dfy

```diff
@@ -708,6 +773,522 @@ module DefaultKeyStorageInterface {
             LogicalName := logicalKeyStoreNameUtf8
           ));
     }
+
+    // This a Get for 3 items, with ConsistentRead
+    // From DDB Docs: "If a requested item does not exist, it is not returned in the result."
+    // Thus, if Mutation Lock is not returned, than it does not exist.
+    method GetItemsForInitializeMutation' ( input: Types.GetItemsForInitializeMutationInput )
+      returns (output: Result<Types.GetItemsForInitializeMutationOutput, Types.Error>)
+      requires  ValidState()
+      modifies Modifies - {History}
+      // Dafny will skip type parameters when generating a default decreases clause.
+      decreases Modifies - {History}
+      ensures ValidState()
+      ensures GetItemsForInitializeMutationEnsuresPublicly(input, output)
+      ensures unchanged(History)
+    {
+      /** Construct & Issue DDB Request */
+      var mLockKey: DDB.Key := map[
+        Structure.BRANCH_KEY_IDENTIFIER_FIELD := DDB.AttributeValue.S(input.Identifier),
+        Structure.TYPE_FIELD := DDB.AttributeValue.S(Structure.MUTATION_LOCK_TYPE)
+      ];
+      var activeKey: DDB.Key := map[
+        Structure.BRANCH_KEY_IDENTIFIER_FIELD := DDB.AttributeValue.S(input.Identifier),
+        Structure.TYPE_FIELD := DDB.AttributeValue.S(Structure.BRANCH_KEY_ACTIVE_TYPE)
+      ];
+      var beaconKey: DDB.Key := map[
+        Structure.BRANCH_KEY_IDENTIFIER_FIELD := DDB.AttributeValue.S(input.Identifier),
+        Structure.TYPE_FIELD := DDB.AttributeValue.S(Structure.BEACON_KEY_TYPE_VALUE)
+      ];
```

Comment:
Ryan and I have agreed to replace Batch Get Item with Transact Get Item.
Ryan has advocated for an option that uses Query;
such an option is a very low priority,
and will probably not be in scope for even GA.

i.e: It will not be implemented unless a customer asks for it.